### DISCUSSION TITLE
Google sheet - loading into a Base

### DISCUSSION THREAD

**Discussion by xu2xulim** [deta/cloud-docs#311] [Help] [answered]
If I am not mistaken, I read that Google api does not seem to work on Deta. My question is what is the best way to get 2000 rows onto a base? This will be triggered by a user request. Related question, will I be a to fetch all 2000 records in one fetch() to iterate through the items?

**Comment by jnsougata** [comment]
`fetch()` certainly has some limitations like record size and to circumvent that,  pagination exists. every page you receive will be having a `last` key which will point to the next page. you have to keep fetching until the last key you receive is `None`

![image](https://user-images.githubusercontent.com/53375272/178085699-309a66d6-9dca-4e2e-9ee0-81cae59884c3.png)

**Comment by abdelhai** [comment, marked as answer]
> My question is what is the best way to get 2000 rows onto a base?

I would try it, it maybe works now if google fixed their issue. by the way, the issue is that google sdks sometimes do not install correctly. you still can install their sdk locally and write a simple python/js script to load your data from base into sheets.

> will I be a to fetch all 2000 records in one fetch() to iterate through the items?

yes if you use pagination. base doesnt limit the number of items, rather the response size, being 1mb per page. but you can fetch as many pages as you want

**Reply by xu2xulim** [reply]
@abdelhai @jnsougata Thank you for your replies. I will marked this as Answered.
